### PR TITLE
implemented drag/drop for tabs

### PR DESCRIPTION
**Pull request by mslourens**
implements Feature Request #842 - drag + drop of snippet tabs

### CODE REVIEW

**Comment by kazup01** [comment]
@mslourens  Cool!

File: browser/components/SnippetTab.js [hunk 1]
```diff
@@ -85,6 +85,19 @@ class SnippetTab extends React.Component {
     })
   }
 
+  handleDragStart (e) {
+    e.dataTransfer.dropEffect = 'move'
+    this.props.onDragStart(e)
+  }
+
+  handleDragEnd (e) {
+    this.props.onDragEnd(e)
```
Comment (kohei-takata) [inline browser/components/SnippetTab.js]: There are no props `onDragEnd`.
Could you delete this function?
Or did you forget to implement?

Comment (mslourens) [reply browser/components/SnippetTab.js]: I started with using onDragStart and onDragEnd, but decided that I needed the onDrop for this job. I removed the handler.

**Comment by mslourens** [comment]
@kazup01 can you have another look at this PR? I made the requested changes.

**Comment by kazup01** [comment]
Thanks @mslourens! Please wait for review from @kohei-takata .

**Comment by kazup01** [comment]
We will check it tommorow :)

File: browser/main/Detail/SnippetNoteDetail.js [hunk 2]
```diff
@@ -234,6 +234,26 @@ class SnippetNoteDetail extends React.Component {
...
+  }
+
+  handleTabDrop (e, index) {
+    const oldIndex = parseInt(e.dataTransfer.getData('text'))
+
+    const snippets = this.state.note.snippets.slice()
+    const draggedSnippet = snippets[oldIndex]
+    snippets[oldIndex] = snippets[index]
+    snippets[index] = draggedSnippet
+    const snippetIndex = index
```
Comment (kohei-takata) [inline browser/main/Detail/SnippetNoteDetail.js]: Could you fix a bug shown in https://gyazo.com/033ea6323d18720dbc108481ca70e6fc ?
It fixes after reload boostnote, so `this.state.note.snippets` will be wrong after drop I think.

Comment (kazup01) [reply browser/main/Detail/SnippetNoteDetail.js]: Hi @mslourens

Comment (mslourens) [reply browser/main/Detail/SnippetNoteDetail.js]: sorry for my late response @kazup01, I was very busy lately. I fixed the bug, I forgot to reload the tab at the oldIndex. Now it's working again.

Comment (kazup01) [reply browser/main/Detail/SnippetNoteDetail.js]: That's ok, thanks for your support ;)

Comment (kazup01) [reply browser/main/Detail/SnippetNoteDetail.js]: Could you review it again? @kohei-takata

**Pull request merged**